src/me.rs

```diff
@@ -97,18 +127,58 @@ mod nasm {
     sum
   }
 
+  #[target_feature(enable = "sse2")]
+  unsafe fn sad_sse2<T: Pixel>(
+    plane_org: &PlaneSlice<'_, T>, plane_ref: &PlaneSlice<'_, T>, blk_h: usize,
+    blk_w: usize
+  ) -> u32 {
+    assert!(mem::size_of::<T>() == 1, "only implemented for u8 for now");
```

Comment:
Uhhhh this is pretty weird. Is there syntax for writing this for just one type? E.g. sad_sse2<T: u8>? @lu-zero ?

Reply:
I can try to look it up a shorter way but moving it to a trait seems the simplest way to achieve what you want.

Reply:
> Is there syntax for writing this for just one type? E.g. sad_sse2<T: u8>?

If the function supports only `u8` temporarily (i.e. it will also support `u16` in the future), this assert seems good.

On the contrary, if it will never support `u16`, it needs not be generic over `T`:

```rust
  unsafe fn sad_sse2_u8(
    plane_org: &PlaneSlice<'_, u8>, plane_ref: &PlaneSlice<'_, u8>, blk_h: usize,
    blk_w: usize
  ) -> u32 {
    // …
  }
```

However, it is not possible to select the implementation based on the parameter type, like in C++:

```cpp
#include <iostream>

template <typename T>
void print(T value);

template <>
void print(int value) {
    std::cout << "int: " << value << std::endl;
}

template <>
void print(double value) {
    std::cout << "double: " << value << std::endl;
}

int main() {
    print(42);
    print(42.);
}
```

```
$ ./test
int: 42
double: 42
```

The selection of the correct implementation based on the generic type can be done via an explicit test (e.g. `mem::size_of::<T>()`, or via additional methods on the `Pixel` trait).